nucypher/cli/worklock.py

```diff
@@ -0,0 +1,132 @@
+from nucypher.characters.banners import WORKLOCK_BANNER
+from nucypher.cli.actions import select_client_account
+from nucypher.cli.config import nucypher_click_config
+from nucypher.cli.painting import paint_receipt_summary, paint_worklock_status, paint_worklock_participant_notice
+from nucypher.cli.types import EIP55_CHECKSUM_ADDRESS, WEI, EXISTING_READABLE_FILE
+
+
+@click.command()
+@click.argument('action')
+@click.option('--force', help="Don't ask for confirmation", is_flag=True)
+@click.option('--sync/--no-sync', default=False)
+@click.option('--poa', help="Inject POA middleware", is_flag=True, default=False)
+@click.option('--provider', 'provider_uri', help="Blockchain provider's URI", type=click.STRING, default="auto://")
+@click.option('--registry-filepath', help="Custom contract registry filepath", type=EXISTING_READABLE_FILE)
+@click.option('--bidder-address', help="Bidding address.", type=EIP55_CHECKSUM_ADDRESS)
+@click.option('--value', help="Bid amount in wei", type=WEI)
+@nucypher_click_config
+def worklock(click_config, action, force, provider_uri, sync, registry_filepath, poa, bidder_address, value):
+    """Participate in NU token worklock bidding."""
+
+    emitter = click_config.emitter
+    click.clear()
+    emitter.banner(WORKLOCK_BANNER)
+
+    try:
+
+        #
+        # Connect to Blockchain and Registry
+        #
+        if not BlockchainInterfaceFactory.is_interface_initialized(provider_uri=provider_uri):
+            BlockchainInterfaceFactory.initialize_interface(provider_uri=provider_uri, sync=sync, poa=poa)
+        blockchain = BlockchainInterfaceFactory.get_interface(provider_uri=provider_uri)
+        if registry_filepath:
+            registry = LocalContractRegistry(filepath=registry_filepath)
+        else:
+            registry = InMemoryContractRegistry.from_latest_publication()
+
+        # Make Agent
+        WORKLOCK_AGENT = ContractAgency.get_agent(WorkLockAgent, registry=registry)
+
+        #
+        # Read-Only Actions
+        #
+
+        if action == "status":
+            paint_worklock_status(emitter=emitter, registry=registry)
+            return  # Exit
+
+        #
+        # Action Switch with Bid address
+        #
+
+        if not bidder_address:
+            bidder_address = select_client_account(emitter=emitter, provider_uri=provider_uri)
+            if force:
+                raise click.MissingParameter("Missing --bidder-address.")
+
+        if action == "remaining-work":
+            remaining_work = WORKLOCK_AGENT.get_remaining_work(target_address=bidder_address)
+            emitter.message(f"Work Remaining for {bidder_address}: {remaining_work}")
+            return  # Exit
+
+        #
+        # Authenticated Action Switch
+        #
+
+        if action == "bid":
+            if not value:
+                value = int(Web3.fromWei(click.prompt("Enter bid amount in ETH", type=click.FloatRange(min=0)), 'wei'))
+                if force:
```

Comment:
same issue of ordering of `force` logic as described above? Unless I'm missing something...

Reply:
Ignore.